src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -191,7 +194,7 @@ export default class DefaultMessagingSession implements MessagingSession {
       !this.isClosing &&
       this.canReconnect(event.code) &&
       this.reconnectController.retryWithBackoff(async () => {
-        this.startConnecting(true);
+        await this.startConnecting(true);
```

Comment:
Yes, but as per this discussion: https://github.com/aws/amazon-chime-sdk-js/pull/2179/files#r860311122, the `await` does not gurantee that the first start is successful, the start being successful is only guranteed when the messagingSessionDidStart is triggered.

Reply:
`messagingSessionDidStart` only triggers if successful.   The await here is to make sure any errors bubble up.  Not to await for a success.  The await might not be strictly necessary but it doesn't hurt I don't think and it might be necessary